core/src/main/java/io/aiven/klaw/service/TopicControllerService.java

```diff
@@ -708,6 +709,16 @@ public ApiResponse approveTopicRequests(String topicId) throws KlawException {
     int tenantId = commonUtilsService.getTenantId(userName);
     TopicRequest topicRequest = getTopicRequestFromTopicId(Integer.parseInt(topicId), tenantId);
 
+    String needOfExtraPermissionForPromote =
+        manageDatabase.getKwPropertyValue(KLAW_EXTRA_PERMISSION_TOPIC_PROMOTION_KEY, tenantId);
+    if (topicRequest.getRequestOperationType().equals(RequestOperationType.PROMOTE.value)
```

Comment:
This means that if a user also has the CreateTopic Permission they will also require the optional promote permission.

Should it be in the check that they have either the Create or Promote version? 

This is just a question though on the UX I don't mind which way its implemented as long as we are happy.

Reply:
Valid question. Approve_topics permission is always required. But if they set this server config property to true, then approve_topics_promotion check is validated